src/chess/board.cc

```diff
@@ -450,46 +447,42 @@ MoveList ChessBoard::GeneratePseudolegalMoves() const {
         result.emplace_back(source, destination);
       }
       // Castlings.
-      if (castlings_.we_can_00()) {
-        bool can_castle = true;
-        for (int i = 5; i < 7; ++i) {
-          if (our_pieces_.get(i) || their_pieces_.get(i)) {
-            can_castle = false;
-            break;
-          }
+      auto walk_free = [this](int from, int to, int rook, int king) {
+        for (int i = from; i <= to; ++i) {
+          if (i == rook || i == king) continue;
+          if (our_pieces_.get(i) || their_pieces_.get(i)) return false;
```

Comment:
The following seems to be marginally faster, but probably not worth it:
```
        uint64_t mask = (2 << to) - (1 << from) - (1 << rook) - (1 << king);
        return !(mask & (our_pieces_.as_int() | their_pieces_.as_int()));
```

Reply:
Maybe it worth it, but I don't understand how it works. Will look again tomorrow. :)

Reply:
Before simplification, the left part was `((1 << (to + 1)) - 1) - ((1 << from) - 1)` to make a bitmask with bits from `from` to `to` set, and the right part clears the bits for rook and king.